public/scripts/extensions/connection-manager/index.js

```diff
@@ -0,0 +1,494 @@
+import { main_api, saveSettingsDebounced } from '../../../script.js';
+import { extension_settings, renderExtensionTemplateAsync } from '../../extensions.js';
+import { callGenericPopup, Popup, POPUP_TYPE } from '../../popup.js';
+import { executeSlashCommandsWithOptions } from '../../slash-commands.js';
+import { SlashCommand } from '../../slash-commands/SlashCommand.js';
+import { ARGUMENT_TYPE, SlashCommandArgument } from '../../slash-commands/SlashCommandArgument.js';
+import { enumIcons } from '../../slash-commands/SlashCommandCommonEnumsProvider.js';
+import { enumTypes, SlashCommandEnumValue } from '../../slash-commands/SlashCommandEnumValue.js';
+import { SlashCommandParser } from '../../slash-commands/SlashCommandParser.js';
+import { uuidv4 } from '../../utils.js';
+
+const MODULE_NAME = 'connection-manager';
+const NONE = '<None>';
+
+const DEFAULT_SETTINGS = {
+    profiles: [],
+    selectedProfile: null,
+};
+
+const COMMON_COMMANDS = [
+    'api',
+    'preset',
+    'api-url',
+    'model',
+];
+
+const CC_COMMANDS = [
+    ...COMMON_COMMANDS,
+    'proxy',
+];
+
+const TC_COMMANDS = [
+    ...COMMON_COMMANDS,
+    'instruct',
+    'context',
+    'instruct-state',
+    'tokenizer',
+];
+
+const FANCY_NAMES = {
+    'api': 'API',
+    'api-url': 'Server URL',
+    'preset': 'Settings Preset',
+    'model': 'Model',
+    'proxy': 'Proxy Preset',
+    'instruct-state': 'Instruct Mode',
+    'instruct': 'Instruct Template',
+    'context': 'Context Template',
+    'tokenizer': 'Tokenizer',
+};
+
+/** @type {() => SlashCommandEnumValue[]} */
+const profilesProvider = () => [
+    new SlashCommandEnumValue(NONE, NONE),
+    ...extension_settings.connectionManager.profiles.map(p => new SlashCommandEnumValue(p.name, p.name, enumTypes.name, enumIcons.server)),
```

Comment:
No need to make this multiline if the name and display name are the same value. You can just set the second argument to `null`, looks cleaner.